deephub/trainer/__init__.py

```diff
@@ -165,12 +169,24 @@ def train(self,
                 hooks=hooks
             )
 
```

Comment:
@nanton96 I don't clearly understand this. We have a flag in order to enable or disable the best_export functionality. But as I can see, in all cases the exporter list will have at least a BestCheckpointCopier object. Also, I don't understand the fact that if we have use_best_exporter enabled, we extend the list with a FinalExporter object. Finally, with this way we cannot select different types of exporter. Also, if we decide to do it, we should add more arguments flag in train function. 

They way I would do it is:
1. Add a string argument flag: exporter_flag: [String]: This will be an enum of {BEST, LAST etc...}
2. Make an ExporterFactory class, with __init__(exporter_type: String) and if statements inside the body, by returning the requested exporter_type. Like if exporter == "best" return BestExporter().
3. Make different implementations of exporters, as different classes as is now.

With this approach, all the code will be removed from inside train, and we will just call ExportersFactor.get_exporter(exporter_type, the object which has the serving_input_receiver_fn). 

@nanton96 WDYT?